docs/customization/controller.rst

```diff
@@ -0,0 +1,142 @@
+    **Frontend Controllers** - these may use many entities at once, they are useful on more general pages. The best example here is the HomepageController that may return many different objects, such as products, banners, blog posts etc.
+
+
+Why would you customize a Controller?
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+To add your custom actions you need to override controllers. You may bee needing to:
+
+* modify your Homepage - add your custom resources, or change template and so on,
+* get a bestsellers list and render it,
+* get a list of customers that have bought some product,
+
+How to customize a Resource Controller?
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Imagine that you would want to render a list of best selling products in a partial template that will be reusable anywhere.
+Assuming that you already have a method on the ``ProductRepository`` (you can see such an example :doc:`here </customization/repository>`.
+Having this method you may be rendering its result in a new action of the ``ProductController`` using a partial template.
+
+See example below:
+
+1. Create a new Controller class under the ``AppBundle/Controller`` namespace.
+
+Remember that it has to extend a proper base class. How can you check that?
+
+For the ``ProductController`` run:
+
+.. code-block:: bash
+
+    $ php app/console debug:container sylius.controller.product
+
+As a result you will get the ``Sylius\Bundle\CoreBundle\Controller\ProductController`` - this is the class that you need to be extending.
+
+Now you have to create the controller that will have an action for rendering a partial template with your bestselling products.
+
+.. code-block:: php
+
+    <?php
+
+    namespace AppBundle\Controller;
+
+    use Sylius\Bundle\CoreBundle\Controller\ProductController as BaseProductController;
+    use Sylius\Component\Core\Repository\ProductRepositoryInterface;
+    use Symfony\Component\HttpFoundation\Response;
+
+    /**
+     * @author Name Surname <name.surname@test.com>
+     */
+    class ProductController extends BaseProductController
+    {
+        /**
+         * @return Response
+         */
+        public function bestsellersAction()
+        {
+            /** @var ProductRepositoryInterface $productRepository */
+            $productRepository = $this->get('sylius.repository.product');
+
+            $bestsellers = $productRepository->findBySold(4);
+
+            return $this->render('SyliusWebBundle:Frontend/Homepage:_advantages.html.twig', ['bestsellers' => $bestsellers]);
+        }
+    }
+
+2. In order to use your controller and its actions you need to configure it in the ``app/config/config.yml``.
+
+.. code-block:: yaml
+
+    sylius_product:
+        resources:
+            product:
+                classes:
+                    controller: AppBundle\Controller\ProductController
```

Comment:
Unfortunately this example is not good, because this kind of thing should be done with existing ``indexAction`` and YAML #Magic (tm). So here I'd show something more advanced or something that cannot be done with default actions.